Bricks support

BernhardSchlegel opened the pull request:
This enables the iSpindle to be added as a "Brick" in [BierBot Bricks](https://bricks.bierbot.com/) allowing you to "attach" the temperature sensor and (more importantly) the **gravity measurement** to your fermenter by drag and dropping!

Screenshot of Fermentation Tracking.

![pr_inspindel](https://user-images.githubusercontent.com/5735721/121638920-0d29af80-ca8c-11eb-8820-84a9d234c3cd.png)

Questions: 
1. I've bumped the version to 7.1.0. (feature added) and not 7.0.1. (minor, bugfix) - is that OK?
2. I've also added the 7.0.0 to the README.MD in the root dir - is that OK?
3. How is the order of supported transmission protocols determined (dropdown in the configuration portal)? 
4. I've noticed that the iSpindel sometimes reports "0.0°P" readings to my backend. I've opened a [issue](https://github.com/universam1/iSpindel/issues/508) in that mattter. Hints appreciated!

Let me know what you think! Happy to incorporate your feedback.

Thanks in advance!

universam1 commented on pio/lib/Sender/Sender.cpp:
```diff
@@ -720,3 +720,175 @@ bool SenderClass::sendBrewblox(String server, uint16_t port, String topic, Strin
+
+    /*
+
+    // forge URL for GET request
```
what is the purpose of all these commented lines, can we clean them up?

BernhardSchlegel replied:
yes, I can (and will do tonight).

BernhardSchlegel replied:
done

universam1 reviewed (changes requested):
Great work, appreciating the README and Version bump as well.

BernhardSchlegel commented:
Hi ! I've removed the unnecessary code and added documentation how to set up. Question 3 remains (how is the order of services in the captive portal defined). Despite that: I'm ready to merge if you want :) 

Big thanks for the awesome project!

Fun fact: I've used the iSpindel you built back from Dez 2016 to test & develop. :)

pppedrillo commented on pio/lib/WiFiManagerKT/WiFiManagerKT.h:
```diff
@@ -56,7 +56,8 @@ var lAPI = [
 {"name":"AWSIOTMQTT", "token":0,"server":1,"uri":1,"port":1,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":1,"hassio":0},
-{"name":"HTTPS Post", "token":1,"server":1,"uri":1,"port":0,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":0,"hassio":0}
+{"name":"HTTPS Post", "token":1,"server":1,"uri":1,"port":0,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":0,"hassio":0},
+{"name":"Bricks (free & easy)",     "token":1,"server":0,"uri":0,"port":0,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":0}];
```
This line breaks Web UI of the portal. Please remove `];` at the end of this line.

BernhardSchlegel replied:
will do!

BernhardSchlegel commented:
Decreased `StaticJsonDocument` to 512 to mitigate memory issues.

pppedrillo commented:
Have you tried to use low level API from WifiClient to send  >1KB payload, without HTTPClient? 
There is info on Arduino forums that this is working well, while HTTPClient does not.
POST pretty much the same way as in `bool SenderClass::sendUbidots(String token, String name)` for example?

BernhardSchlegel commented:
Nope, I did not since Ubidots is using `HTTP` - do you think it works for `HTTPS`? 

And again, quoting from the [issue](https://github.com/universam1/iSpindel/issues/508#issuecomment-861613697): The already existing HTTPS Post service option (as offered by the iSpindel today) is not working using the 1024 Byte large buffer.

BernhardSchlegel commented:
I have (unsucessfully) tested it, see [this branch](https://github.com/BernhardSchlegel/iSpindel/tree/bricks-support-test-secureclient) - I'm unable to connect to my server. Presumably the `WiFiClientSecure` cannot handle HTTPS redirects. If someone can provide guidance I happy to try it again (I really like the simplicity!). If not, lowering the `StaticJsonDocument` buffer to 512 is IMHO our best option (also fixing the "default HTTPS POST not working"-Bug).

lbussy commented:
I'm on my work laptop at the moment, but in the ESP32 core, there's a "follow redirects" option.  I further remember that being added to match the ESP8266 so I'd check the HTTPClient lib and see if it's there.

BernhardSchlegel commented:
Hi @lbussy , any update would be highly appreciated! I looked at the `WiFiClientSecure` [source](https://github.com/espressif/arduino-esp32/blob/master/libraries/WiFiClientSecure/src/WiFiClientSecure.cpp) and was not able to find anything "forward" or "redirect" related. I really love to merge the Bricks support! (And this PR is theoretically ready for merge (and also fixes the HTTP POST Bug))

universam1 commented on pio/lib/WiFiManagerKT/WiFiManagerKT.h:
```diff
@@ -56,7 +56,8 @@ var lAPI = [
 {"name":"AWSIOTMQTT", "token":0,"server":1,"uri":1,"port":1,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":1,"hassio":0},
-{"name":"HTTPS Post", "token":1,"server":1,"uri":1,"port":0,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":0,"hassio":0}
+{"name":"HTTPS Post", "token":1,"server":1,"uri":1,"port":0,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":0,"hassio":0},
+{"name":"Bricks (free & easy)", "token":1,"server":0,"uri":0,"port":0,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":0,"hassio":0}
```
```suggestion
{"name":"BierBot Bricks", "token":1,"server":0,"uri":0,"port":0,"channel":0,"db":0,"username":0,"password":0,"job":0,"instance":0,"warning1":0,"hassio":0}
```
would it be more specific that way

BernhardSchlegel replied:
done !

universam1 reviewed (changes requested):
some small style concerns

universam1 reviewed (approved):
👍🏻

lbussy commented:
I don't do as much with the esp8266, but the code in HTTPClient for the ESP32 is [here](https://github.com/espressif/arduino-esp32/blob/21947ebe76cf6380fe06aa885390b86cc477709a/libraries/HTTPClient/src/HTTPClient.cpp#L1426-L1429).
